api/controllers/game/spectate/join.js

```diff
@@ -44,15 +49,31 @@ module.exports = async function (req, res) {
     }
 
     const { unpackGamestate, createSocketEvent } = sails.helpers.gameStates;
-    const gameState = unpackGamestate(game.gameStates.at(-1));
+    // Default to first gamestate for finished games, last for live ones
+    const gameStateIndex = Number(
+      req.query.gameStateIndex ?? (
+        [ GameStatus.FINISHED, GameStatus.ARCHIVED ]
+          .includes(game.status) ? 0 : -1
+      )
+    );
```

Comment:
This is kindof a chonker of an expression. Not sure how clearly it reads. Intent is to take the query param value if present, then otherwise coalesce to 0 if the game is finished and -1 if it's ongoing

Reply:
Idk if this is any better, The original isn't terrible to read but it is a chunky boi

```suggestion
    const gameStateIndex = Number(req.query.gameStateIndex )?? game.status >= GameStatus.finished ? 0 : -1  
```